### PR TITLE
[gui] GGUI 15/n: Python-side code

### PR DESCRIPTION
**Pull request by AmesingFlank**
Related issue = #2646 

This is the 15th of a series of PRs that adds a GPU-based GUI to taichi.

This PR adds the python-side code. Mostly these are just wrappers around the types exported from pybind. The only pieces of non-trivial code are:

* in `scene.py`: a taichi kernel that computes vertex normals if they're not provided
* in `camera.py`: a FPS-game style user input handling.

### CODE REVIEW

**Review by g1n0st — APPROVED** [review]

File: python/taichi/ui/canvas.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from taichi.core import ti_core as _ti_core
+from taichi.core.primitive_types import *
```
Comment (k-ye) [inline python/taichi/ui/canvas.py:2]: nit

File: python/taichi/__init__.py [hunk 2]
```diff
@@ -11,6 +11,8 @@
 # Issue#2223: Do not reorder, or we're busted with partially initialized module
 from taichi import aot  # isort:skip
 
+from taichi.ui import *  # isort:skip
```
Comment (k-ye) [inline python/taichi/__init__.py:14]: nit: Let's not do this now

**Review by k-ye — COMMENTED** [review]
Thanks, there are a few wildcard imports. Could you remove them?

**Comment by AmesingFlank** [comment]
> Thanks, there are a few wildcard imports. Could you remove them?
sure.

File: python/taichi/__init__.py (hunk 2 above)
Comment (AmesingFlank) [reply python/taichi/__init__.py:14]: Are you referring to the * or the isort:skip or the import itself?

Comment (k-ye) [reply python/taichi/__init__.py:14]: `import *`. It's fine to use `ti.ui`

Comment (AmesingFlank) [reply python/taichi/__init__.py:14]: Ah of course. This is not what I intended to do 😅 will fix.

**Pull request merged**